src/app/3d/qgslightswidget.cpp

```diff
@@ -0,0 +1,126 @@
+  connect( spinPositionX, qgis::overload<double>::of( &QDoubleSpinBox::valueChanged ), this, &QgsLightsWidget::updateCurrentLightParameters );
+  connect( spinPositionY, qgis::overload<double>::of( &QDoubleSpinBox::valueChanged ), this, &QgsLightsWidget::updateCurrentLightParameters );
+  connect( spinPositionZ, qgis::overload<double>::of( &QDoubleSpinBox::valueChanged ), this, &QgsLightsWidget::updateCurrentLightParameters );
+  connect( spinIntensity, qgis::overload<double>::of( &QDoubleSpinBox::valueChanged ), this, &QgsLightsWidget::updateCurrentLightParameters );
+  connect( btnColor, &QgsColorButton::colorChanged, this, &QgsLightsWidget::updateCurrentLightParameters );
+  connect( spinA0, qgis::overload<double>::of( &QDoubleSpinBox::valueChanged ), this, &QgsLightsWidget::updateCurrentLightParameters );
+  connect( spinA1, qgis::overload<double>::of( &QDoubleSpinBox::valueChanged ), this, &QgsLightsWidget::updateCurrentLightParameters );
+  connect( spinA2, qgis::overload<double>::of( &QDoubleSpinBox::valueChanged ), this, &QgsLightsWidget::updateCurrentLightParameters );
+}
+
+void QgsLightsWidget::setPointLights( const QList<QgsPointLightSettings> &pointLights )
+{
+  mPointLights = pointLights;
+  updateLightsList();
+  cboLights->setCurrentIndex( 0 );
+  onCurrentLightChanged( 0 );
+}
+
+QList<QgsPointLightSettings> QgsLightsWidget::pointLights()
+{
+  return mPointLights;
+}
+
+void QgsLightsWidget::onCurrentLightChanged( int index )
+{
+  if ( index < 0 || index >= cboLights->count() )
+    return;
+
+  QgsPointLightSettings light = mPointLights.at( index );
+  whileBlocking( spinPositionX )->setValue( light.position().x() );
+  whileBlocking( spinPositionY )->setValue( light.position().y() );
+  whileBlocking( spinPositionZ )->setValue( light.position().z() );
+  whileBlocking( btnColor )->setColor( light.color() );
+  whileBlocking( spinIntensity )->setValue( light.intensity() );
+  whileBlocking( spinA0 )->setValue( light.constantAttenuation() );
+  whileBlocking( spinA1 )->setValue( light.linearAttenuation() );
+  whileBlocking( spinA2 )->setValue( light.quadraticAttenuation() );
+}
+
+void QgsLightsWidget::updateCurrentLightParameters()
+{
+  int index = cboLights->currentIndex();
+  if ( index < 0 || index >= cboLights->count() )
+    return;
+
+  QgsPointLightSettings light;
+  light.setPosition( QgsVector3D( spinPositionX->value(), spinPositionY->value(), spinPositionZ->value() ) );
+  light.setColor( btnColor->color() );
+  light.setIntensity( spinIntensity->value() );
+  light.setConstantAttenuation( spinA0->value() );
+  light.setLinearAttenuation( spinA1->value() );
+  light.setQuadraticAttenuation( spinA2->value() );
+  mPointLights[index] = light;
+}
+
+void QgsLightsWidget::onAddLight()
+{
+  if ( mPointLights.count() >= 8 )
+  {
+    QMessageBox::warning( this, tr( "Add light" ), tr( "It is not possible to add more than 8 lights to the scene." ) );
+    return;
+  }
+
+  mPointLights << QgsPointLightSettings();
```

Comment:
The default light we use could be probably moved to e.g. 100km above zero and we should be still fine because it has only constant attenuation, so its intensity is everywhere the same.

Reply:
Sounds good.

Reply:
Now I realize that will make rendering of buildings more ugly because the walls will get mostly uniform color due to the light being so far away... probably we should have something auto-adjusted based on scene size... maybe even more lights

Reply:
Maybe scene max( width , height ) / 2 or something, that kind of works for the large scene I have here :)